build/build_image.py

```diff
@@ -0,0 +1,98 @@
+    default_env = os.getenv("IMAGE_" + env, default)
+    if default_env == "":
+        default_env = default
+    env_mid_name = utils.underscore_uppercase(name)
+    
+    env = os.getenv("IMAGE_%s_%s" % (env_mid_name, env), default_env)
+    if env == "":
+        env = default_env
+    return env
+
+def get_image_project(name):
+    return get_image_env(name, "PROJECT", "pingcap")
+
+def get_image_registry(name):
+    return get_image_env(name, "REGISTRY", "localhost:5000")
+
+def get_image_tag(name):
+    return get_image_env(name, "TAG", "latest")
+
+def get_image_build(name):
+    print(get_image_env(name, "BUILD", "1"))
+    return get_image_env(name, "BUILD", "1")
+
+def get_image_full_name(name):
+    project = get_image_project(name)
+    registry = get_image_registry(name)
+    tag = get_image_tag(name)
+
+    return "%s/%s/%s:%s" % (registry, project, name, tag)
+
+def pass_env_to_build_arg(cmd, arg_name):
+    if os.getenv(arg_name) != None:
+        cmd += ["--build-arg", "%s=%s" % (arg_name, os.getenv(arg_name))]
+
+if __name__ == '__main__':
+    cmdParser = argparse.ArgumentParser(description='Helper script to build Chaos Mesh image.')
+    cmdParser.add_argument('name', metavar="NAME", type=str, nargs=1, help="the name of image")
+    cmdParser.add_argument('path', metavar="PATH", type=str, nargs=1, help="the path of the Dockerfile build directory")
+
+    args = cmdParser.parse_args()
+    name = args.name[0]
+    image_full_name = get_image_full_name(name)
+
+    env = {}
+    cmd = []
+    if get_image_build(name) == "1":
+        if os.getenv("DOCKER_CACHE") == "1":
+            env = {"DOCKER_BUILDKIT": "1", "DOCKER_CLI_EXPERIMENTAL": "enabled"}
+            cache_dir = os.getenv("DOCKER_CACHE_DIR", "%s/.cache/image-%s" % (os.getcwd(), name))
+            pathlib.Path(cache_dir).mkdir(parents=True, exist_ok=True)
+            cmd = ["docker", "buildx", "build", "--load", "--cache-to", "type=local,dest=%s" % cache_dir]
+            if os.getenv("DISABLE_CACHE_FROM") != "1":
+                cmd += ["--cache-from", "type=local,src=%s" % cache_dir]
+        else:
+            if os.getenv("TARGET_PLATFORM") != None:
+                env = {"DOCKER_BUILDKIT": "1"}
+                cmd = ["docker", "buildx", "build", "--load", "--platform", os.getenv("TARGET_PLATFORM")]
+            else:
+                # This branch is split to avoid to use `buildx`, as `buildx` is not supported on some CI environment
+                env = {"DOCKER_BUILDKIT": "1"}
+                cmd = ["docker", "build"]
+
+        for env_key in common.export_env_variables:
+            pass_env_to_build_arg(cmd, env_key)
+
+        target_platform = utils.get_target_platform()
+        cmd += ["--build-arg", "%s=%s" % ("TARGET_PLATFORM", target_platform)]
+        cmd += ["-t", image_full_name, args.path[0]]
+    else:
+        cmd = ["docker", "pull", image_full_name]
+
+    print(" ".join(cmd))
+    subprocess.run(cmd, env=env)
```

Comment:
I try to run `make install.sh` in my local machine, and it returns error `FileNotFoundError: [Errno 2] No such file or directory: 'docker'`.

I think `env=env` leads to this problem, maybe the `PATH` environment is covered, and I did a little verification:
```
>>> import subprocess
>>> cmd = ["docker", "images"]
>>> subprocess.run(cmd)
REPOSITORY                                  TAG           IMAGE ID       CREATED        SIZE
localhost:5000/chaos-mesh/chaos-mesh        latest        9bded8a51de2   2 days ago     61.2MB
.....
CompletedProcess(args=['docker', 'images'], returncode=0)
>>> env={}
>>> subprocess.run(cmd,env=env)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python@3.9/3.9.1_2/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 501, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/local/Cellar/python@3.9/3.9.1_2/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 947, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/Cellar/python@3.9/3.9.1_2/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 1819, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'docker
```

Reply:
I modified the `env = {}` to `env = os.environ.copy()`. Does it fix this problem?

Reply:
It works now!